server/routes/prisons/incentiveGroups/addIncentiveGroupController.ts

```diff
@@ -0,0 +1,72 @@
+import { RequestHandler } from 'express'
+import { ValidationChain, body, validationResult } from 'express-validator'
+import { PrisonService, IncentiveGroupService } from '../../../services'
+import { CreateIncentiveGroupDto } from '../../../data/visitSchedulerApiTypes'
+import incentiveLevels from '../../../constants/incentiveLevels'
+
+export default class AddIncentiveGroupController {
+  public constructor(
+    private readonly prisonService: PrisonService,
+    private readonly incentiveGroupService: IncentiveGroupService,
+  ) {}
+
+  public view(): RequestHandler {
+    return async (req, res) => {
+      const { prisonId } = req.params
+      const prison = await this.prisonService.getPrison(res.locals.user.username, prisonId)
+      const formValues = req.flash('formValues')?.[0] || {}
+
+      res.render('pages/prisons/incentiveGroups/addIncentiveGroup', {
+        errors: req.flash('errors'),
+        prison,
+        formValues,
+        incentiveLevels,
+      })
+    }
+  }
+
+  public submit(): RequestHandler {
+    return async (req, res) => {
+      const { prisonId } = req.params
+
+      const originalUrl = `/prisons/${prisonId}/incentive-groups/add`
+
+      const errors = validationResult(req)
+      if (!errors.isEmpty()) {
+        req.flash('errors', errors.array())
+        req.flash('formValues', req.body)
+        return res.redirect(originalUrl)
+      }
+
+      const createIncentiveGroupDto: CreateIncentiveGroupDto = {
+        name: req.body.name,
+        prisonId,
+        incentiveLevels: req.body.incentiveLevels,
+      }
+      try {
+        const { reference } = await this.incentiveGroupService.createIncentiveGroup(
+          res.locals.user.username,
+          createIncentiveGroupDto,
+        )
+        req.flash('message', `Location group '${reference}' has been created`)
```

Comment:
Missed this earlier; this message isn't being picked up by the 'View single' page - and text should be 'Incentive level' instead of 'Location'